### PR TITLE
fix(test): update RTL test example

### PR DESCRIPTION
**Pull request by horacioh**
update the React Testing Library tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,5 +54,8 @@
     "vitepress": "^0.21.4",
     "vitest": "workspace:*",
     "vue": "^3.2.25"
+  },
+  "dependencies": {
+    "test": "^0.6.0"
```
Comment (antfu) [inline package.json:59]: what's is this for?

File: packages/ui/client/components/DetailsPanel.vue [hunk 2]
```diff
@@ -8,7 +8,17 @@ const open = ref(true)
 
 <template>
   <div :open="open" class="details-panel" @toggle="open = $event.target.open">
-    <div p="y1" text-sm items-center gap-2 :class="color" w-full flex select-none cursor-pointer>
```
Comment (antfu) [inline packages/ui/client/components/DetailsPanel.vue:11]: Can you revert these format changes?

**Review by antfu — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (horacioh) [reply package.json]: uops! my bad!

File: packages/ui/client/components/DetailsPanel.vue (hunk 2 above)
Comment (horacioh) [reply packages/ui/client/components/DetailsPanel.vue]: done!

**Comment by antfu** [comment]
Please keep unrelated files unchanged, thanks

**Comment by horacioh** [comment]
hey @antfu !

do you prefer I redo the PR just by changing the example code? let me know what do you think is best!

thanks!

**Comment by antfu** [comment]
I don't mind how you do it, but it seems

- Your formatter is formating files under `/ui`
- You used `yarn` which we are using `pnpm`

**Comment by dammy001** [comment]
Yeah... @antfu i think you should add preinstall hook to allow only pnpm 🤔

File: examples/react-testing-lib/src/App.test.tsx [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-    expect(screen.getByText(/Hello Vite \+ React!/i)).toBeDefined()
+    expect(screen.getByText(/Hello Vite \+ React!/i)).toBeInTheDocument()
   })
 
-  it('should increment count on click', () => {
-    render(<App />)
-    userEvent.click(screen.getByRole('button'))
-    expect(screen.findByText(/count is: 1/i)).toBeDefined()
+  it('should increment count on click', async() => {
+    await render(<App />)
```
Comment (mitchelvanbever) [inline examples/react-testing-lib/src/App.test.tsx:12]: No need to await the render

File: examples/react-testing-lib/src/App.test.tsx [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
+    const btn = await screen.findByRole('button')
```
Comment (mitchelvanbever) [inline examples/react-testing-lib/src/App.test.tsx:13]: also no need to await this

File: examples/react-testing-lib/src/App.test.tsx [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
+    expect(screen.getByText(/count is: 0/i))
+    userEvent.click(btn)
+    await waitFor(() => {
```
Comment (mitchelvanbever) [inline examples/react-testing-lib/src/App.test.tsx:16]: Now here you probably want to await but preferably without waitFor

File: examples/react-testing-lib/src/App.test.tsx [hunk 6]
```diff
@@ -1,16 +1,20 @@
-import { describe, expect, it } from 'vitest'
-import { render, screen, userEvent } from './utils/test-utils'
+import '@testing-library/jest-dom'
```
Comment (mitchelvanbever) [inline examples/react-testing-lib/src/App.test.tsx:1]: Probably best to use a setupFile

**Review by mitchelvanbever — CHANGES_REQUESTED** [review]
Thanks for checking it out. Left some comments and I do think there is bigger problem at play here.

Whenever you include @testing-library/jest-dom the types for all vitest related stuff gets overwritten by @types/jest

**Comment by sheremet-va** [comment]
> Whenever you include @testing-library/jest-dom the types for all vitest related stuff gets overwritten by @types/jest

We have an open issue for this https://github.com/vitest-dev/vitest/issues/517

**Comment by horacioh** [comment]
thanks everyone that review this PR!. I decided to create a new cleaner on with all the suggestions and fixes. you can check it out here #530 . Going to close this one!

Cheers!

**Pull request closed without merging**